### PR TITLE
[MM-24663] Refactor link metadata store to use squirrel

### LINKED ISSUE #14423: Refactor SQL queries in store/sqlstore/link_metadata_store.go to use the squirrel builder
Mattermost is replacing the SQL queries of the store layer to use the [squirrel builder](https://github.com/Masterminds/squirrel). This help wanted issue is to change the usages of raw SQL queries in the file `store/sqlstore/link*metadata*store.go`.

The expected way to implement this is to search in the `store/sqlstore/link*metadata*store.go` file for all the strings containing a SQL query and rewrite them with squirrel. To do so, we must obtain the query builder using the `getQueryBuilder()` method from the store struct, and use it to build the same query that we have in the string.

You can find more information on how to use squirrel in [the official package documentation](https://godoc.org/github.com/Masterminds/squirrel).

Example Pull Request: https://github.com/mattermost/mattermost-server/pull/14393

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` in https://community.mattermost.com/

---
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-24663

### PR DESCRIPTION
**Pull request by dantepippi**
#### Summary

This pull request refactors the link metadata store to use squirrel queries instead of raw SQL queries.

Closes #14423 

Jira Ticket: https://mattermost.atlassian.net/browse/MM-24663

### CODE REVIEW

**Comment by hanzei** [comment]
@mattermost/core-build-engineers The branch seams to exist: https://github.com/dantepippi/mattermost-server/tree/MM-24663_refactor-link-metadata-store-to-use-squirrel

**Comment by dantepippi** [comment]
> @mattermost/core-build-engineers The branch seams to exist: https://github.com/dantepippi/mattermost-server/tree/MM-24663_refactor-link-metadata-store-to-use-squirrel



> @mattermost/core-build-engineers The branch seams to exist: https://github.com/dantepippi/mattermost-server/tree/MM-24663_refactor-link-metadata-store-to-use-squirrel

Greetings @hanzei . Weird thing is that the error about the branch mentions the path "mattermost-**webapp**" while the PR was meant to mattermost-server

File: store/sqlstore/link_metadata_store.go [hunk 1]
```diff
@@ -53,15 +55,16 @@ func (s SqlLinkMetadataStore) Save(metadata *model.LinkMetadata) (*model.LinkMet
...
-		FROM
-			LinkMetadata
-		WHERE
-			URL = :URL
-			AND Timestamp = :Timestamp`, map[string]interface{}{"URL": url, "Timestamp": timestamp})
+	query := s.getQueryBuilder().
+		Select("*").
+		From("LinkMetadata").
+		Where(sq.Eq{"URL": url}).
+		Where(sq.Eq{"Timestamp": timestamp})
```
Comment (mgdelacroix) [inline store/sqlstore/link_metadata_store.go:62]: These two can be merged in a single call to where:

```suggestion
		Where(sq.Eq{"URL": url, "Timestamp": timestamp})
```

**Pull request merged**